src/content/docs/en/guides/cms/cosmic.mdx

```diff
@@ -0,0 +1,322 @@
+
+## Prerequisites
+
+1. **A new Astro project**- If you’d like to start with a fresh Astro project, read the [installation guide](https://docs.astro.build/en/install/auto/). You can also follow along with the template used in this guide by cloning the [Astro Headless CMS Theme](https://astro.build/themes/details/cosmic-cms-blog/). This uses [Tailwind CSS](https://tailwindcss.com/) for some basic styling.
+2. **A Cosmic account and Bucket** - [Create a free Cosmic account](https://app.cosmicjs.com/signup) if you don’t have one yet. After creating your account, you will be prompted to create a new empty project. You can also select the [Simple Astro Blog template](https://www.cosmicjs.com/marketplace/templates/simple-astro-blog) (this is the same template as the Astro Headless CMS Theme) to automatically import all of the content used in this guide.
+
+## Integrating Cosmic with Astro
+
+Let’s start by using the [Cosmic JavaScript SDK](https://github.com/cosmicjs/cosmic-sdk-js) to fetch data from our Cosmic Bucket.
+
+### Configuring API Keys
+
+API keys are your bridge between Cosmic and Astro project. In your Cosmic dashboard in the top navigation menu, navigate to **API keys** to retrieve both your **Bucket slug** and **Bucket read key**. Paste these into your **.env** file in your Astro project.
+
+```ini title=".env"
+PUBLIC_COSMIC_BUCKET_SLUG=YOUR_BUCKET_SLUG
+PUBLIC_COSMIC_READ_KEY=YOUR_READ_KEY
+```
+
+## Installing Necessary Dependencies
+
+Astro's ecosystem works with dependencies that enhance its capabilities. To seamlessly pull content from Cosmic, you'll need to add the [Cosmic JavaScript SDK](https://www.npmjs.com/package/@cosmicjs/sdk). Let’s also add three more packages we will use later, [date-fns](https://www.npmjs.com/package/date-fns) (to format the date), and [react-markdown](https://www.npmjs.com/package/react-markdown) (to render markdown content from Cosmic).
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install @cosmicjs/sdk date-fns react-markdown
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add @cosmicjs/sdk date-fns react-markdown
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add @cosmicjs/sdk date-fns react-markdown
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Since we will be using [react-markdown](https://www.npmjs.com/package/react-markdown) to render the markdown content coming from Cosmic, we will need to add the [Astro React Integration](https://docs.astro.build/en/guides/integrations-guide/react/).
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npx astro add react
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm astro add react
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn astro add react
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Lastly, let’s add the [Astro Image Integration](https://docs.astro.build/en/guides/integrations-guide/image/) for optimized image rendering.
```

Comment:
The image integration is depricated and should be replaced with `astro:assets`. Can you incorporate those changes?

Reply:
Making all of the deprecated image changes now.